### PR TITLE
Rjwebb/conversations list uses summary endpoint

### PR DESCRIPTION
**Pull request by rjwebb**
This PR adds a new endpoint `GET /api/v3/conversations_summary` which returns a list of the conversations with only the fields returned that we need for the "Manage Conversation" and Dashboard conversation lists. It also includes the UI changes needed for that. This includes a new reducer `conversations_summary`.

Some more changes:
- The list of conversations and the filter buttons in the dashboard now live in a separate component `ConversationsList`
- After moderating a conversation, the conversation list is updated
- The single conversation view on the Dashboard uses `zid_metadata`
- Redirect to dashboard main page if conversation cannot be found (e.g. broken link)

### CODE REVIEW

File: server/src/server.ts [hunk 1]
```diff
@@ -7824,6 +7830,62 @@ async function handle_GET_conversations(
...
+    }
+
+    // Make sure zid is not exposed
+    delete conv.zid;
+
+    delete conv.is_anon;
+    delete conv.is_public;
+    if (conv.context === "") {
+      delete conv.context;
+    }
```
Comment (rjwebb) [inline server/src/server.ts:7882]: These delete statements are not needed

File: client/src/actions.tsx [hunk 2]
```diff
@@ -515,7 +516,10 @@ export const populateZidMetadataStore = (conversation_id, reload?: boolean) => {
     dispatch(requestZidMetadata(conversation_id))
     return fetchZidMetadata(conversation_id).then(
       (res) => dispatch(receiveZidMetadata(res)),
-      (err) => dispatch(zidMetadataFetchError(err)),
+      (err) => {
+        console.log(err)
```
Comment (rjwebb) [inline client/src/actions.tsx:520]: Remove this

File: client/src/pages/dashboard/conversations_list.tsx [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+  const conversations = data || []
+
+  const [showAllFIPConversations, setShowAllFIPConversations] = useLocalStorage("showAll", true)
+  const [showOpenFIPConversations, setShowOpenFIPConversations] = useLocalStorage("showOpen", false)
+  const [showNonFIPConversations, setShowNonFIPConversations] = useLocalStorage("showNonFIP", false)
+  const [showArchivedConversations, setShowArchivedConversations] = useLocalStorage(
+    "showArchived",
+    false,
+  )
+  const [showHiddenConversations, setShowHiddenConversations] = useLocalStorage("showHidden", false)
```
Comment (rjwebb) [inline client/src/pages/dashboard/conversations_list.tsx:24]: These selections are mutually exclusive so we could simplify the code a lot here

**Comment by raykyri** [comment]
Looks good to me

**Pull request merged**